### ISSUE TITLE
This browser or app may not be secure.

### ISSUE THREAD

**Issue by NoString** [linouk23/youtube_uploader_selenium#39]
hi, developer. I try to use this python code, but when I login youtube, it notice me that "browser is not secure", how can I do?
my firefox version is 89.0 and had installed and configure the geckodriver
![image](https://user-images.githubusercontent.com/23315597/121806511-ae7a5680-cc82-11eb-9745-3189fa58fe78.png)

**Comment by terahlunah** [comment]
I have the same issue

**Comment by rzfzr** [comment]
You don't log in, just load the cookies, get them with a extension like editThisCookie

 `async function setCookies(channelName) {        let p = require("electron").remote.app.getAppPath();        p = p.substring(0, p.lastIndexOf("\\"));        let cookieJar = JSON.parse(          fs.readFileSync(            p + "/cookieJar/" + (channelName || "Só idéias legais!") + ".json",            "utf8"          )        );        cookieJar.forEach(async (cookie) =&gt; {          cookie.sameSite = "None";          cookie.storeId = "None";          cookie.secure = true;          await driver.manage().addCookie(cookie);        });      }`

**Comment by pokima** [comment]
> You don't log in, just load the cookies, get them with a extension like editThisCookie
> 
> `async function setCookies(channelName) { let p = require("electron").remote.app.getAppPath(); p = p.substring(0, p.lastIndexOf("\\")); let cookieJar = JSON.parse( fs.readFileSync( p + "/cookieJar/" + (channelName || "Só idéias legais!") + ".json", "utf8" ) ); cookieJar.forEach(async (cookie) =&gt; { cookie.sameSite = "None"; cookie.storeId = "None"; cookie.secure = true; await driver.manage().addCookie(cookie); }); }`


I use editthiscookie to get the same cookies for different channels

**Comment by rzfzr** [comment]
@pokima yes, even the same account as manager for multiple channels/brand accounts, you can get different cookies and use them

**Comment by NoNoNe96** [comment]
> You don't log in, just load the cookies, get them with a extension like editThisCookie
> 
> `async function setCookies(channelName) { let p = require("electron").remote.app.getAppPath(); p = p.substring(0, p.lastIndexOf("\\")); let cookieJar = JSON.parse( fs.readFileSync( p + "/cookieJar/" + (channelName || "Só idéias legais!") + ".json", "utf8" ) ); cookieJar.forEach(async (cookie) =&gt; { cookie.sameSite = "None"; cookie.storeId = "None"; cookie.secure = true; await driver.manage().addCookie(cookie); }); }`

How do I get Login Cookies?

**Comment by rzfzr** [comment]
> How do I get Login Cookies?
Did you try the extension? just click the icon > export > paste them

**Comment by gartha1** [comment]
I don't quite understand.. So I run firefox I use that  cookie editor extension, go to youtube while logged in, and click the export button in cookie editor, to copy the cookies.  

So now I have cookie data in my clipboard.

I could put that in a file c:\thispythonproject\cookie.json 

But how do I get this program to load the cookies from that file?

A comment mentions "async function setCookies(channelName) {.........}"

Well, having done a bit of javascript before, I suppose that's javascript rather than python.

Where am  I meant to put that javascript line?

And I suppose straight after that line there should be a line that calls it?

I guess Selenium will run a browser or headless browser, and it's doing that so that's fine, but I suppose there must be a way to inject javascript into a page, and if injecting it into the right page then hey presto it could work.

But could anybody please explain how to do that?

(So I can go from having that cookie data, in a json file, to getting the cookies auto loaded into the browser that the program starts?)

Thanks
